jujugui/static/gui/src/app/jujulib/terms.js

```diff
@@ -113,29 +115,31 @@ var module = module;
         first parameter and an authorization object as its second.
     */
     addAgreement: function(terms, callback) {
-      var self = this;
-      var handler = function(error, response) {
+      const handler = (error, response) => {
         if (error !== null) {
           callback(error, null);
           return;
         }
-        var terms = response.agreements.map(function(term) {
-          return self._formatTerm(term);
-        });
+        const terms = response.agreements.map(term => this._formatTerm(term));
         callback(null, terms);
       };
-      var url = this.url + '/agreement';
-      var payload = terms.map(function(term) {
-        const args = {
-          termname: term.name,
-          termrevision: term.revision
-        };
-        if (term.owner) {
-          args.termowner = term.owner;
-        }
-        return args;
-      });
-      return jujulib._makeRequest(this.bakery, url, 'POST', payload, handler);
+      const url = this.url + '/agreement';
+      const body = JSON.stringify(
+        terms.map(term => {
+          const args = {
+            termname: term.name,
+            termrevision: term.revision
+          };
+          if (term.owner) {
+            args.termowner = term.owner;
+          }
+          return args;
+        }));
+      const headers = {
+        'Content-Type': 'application/json'
```

Comment:
Just to clarify - we have constants on a per file basis, but nothing global to jujulib.